app/controllers/CacheController.scala

```diff
@@ -79,13 +81,15 @@ class CacheController @Inject() (
       }
   }
 
+  private def isRequestTransitional()(implicit request: VersionedRequest[AnyContent]) = request.phase == Phase.Transition
```

Comment:
Consider adding `def isTransitional: Boolean = this == Transition` to the Phase trait

Then in CacheController we can have `case Some(userAnswers) if request.phase.isTransitional == userAnswers.isTransitional =>` and in CacheRepository we can have `Some(Updates.setOnInsert("isTransitional", phase.isTransitional)),`